Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +447,131 @@ def restore_snapshot_online(
         snapshot_name,
         {})
 
+def create_schedule(
+        client_obj,
+        storage_system_ip,        
+        storage_system_username,
+        storage_system_password,
+        schedule_name,
+        snapshot_name,
+        base_volume_name,
+        read_only,
+        expiration_time,
+        retention_time,
+        expiration_unit,
+        retention_unit,
+        task_freq):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "Schedule creation failed. Storage system username or password is \
+null",
+            {})
+    if schedule_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Schedule name is null",
+            {})			
+    if snapshot_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Snapshot name is null",
+            {})
+    if len(schedule_name) < 1 or len(schedule_name) > 31:
+        return (False, False, "Schedule creation failed. Schedule name must be atleast 1 character and not more than 31 characters", {})
+
+    if len(snapshot_name) < 1 or len(snapshot_name) > 31:
+        return (False, False, "Schedule create failed. Snapshot name must be atleast 1 character and not more than 31 characters", {})
+    if base_volume_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Base volume name is null",
+            {})
+    if len(base_volume_name) < 1 or len(base_volume_name) > 31:
+        return (False, False, "Schedule create failed. Base volume name must be atleast 1 character and not more than 31 characters", {})
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        expirationHours = convert_to_hours(expiration_time, expiration_unit)
+	retentionHours = convert_to_hours(retention_time, retention_unit)
+        if client_obj.volumeExists(base_volume_name):
+           cmd = ["\"createsv"] 
```

Comment:
What happens when the schedule that we are creating already exists? We need to check if a schedule is created by that name. Therefore, the following methods need to be added in SDK and client

a. getSchedule(name)
b. scheduleExists(name)

scheduleExists will refer try and get the schedule and based on that result we have to decide in ansible whether to create the new schedule

Reply:
If schedule exist it show error that schedule already exist ..Can we have only one function scheduleExists(name) in sdk